src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy;
+
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> inventory, String fruit, int quantity) {
+        int currentQuantity = inventory.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in inventory.");
+        }
```

Comment:
According to the checklist, you should check the result balance before saving it in the Storage to ensure it is positive. While you are throwing a RuntimeException when the inventory is insufficient, you are not explicitly checking for a non-positive result balance after the operation. Consider adding a check after line 12 to ensure the new quantity is non-negative.